sync_diff_inspector/config.go

```diff
@@ -283,40 +289,56 @@ func (c *Config) checkConfig() bool {
 		return false
 	}
 
-	if len(c.SourceDBCfg) == 0 {
-		log.Error("must have at least one source database")
-		return false
-	}
-
-	for i := range c.SourceDBCfg {
-		if !c.SourceDBCfg[i].Valid() {
+	if len(c.DMAddr) != 0 {
+		u, err := url.Parse(c.DMAddr)
+		if err != nil || u.Scheme == "" || u.Host == "" {
+			log.Error("dm-addr's format should like 'http://127.0.0.1:8261'")
 			return false
 		}
-		if c.SourceDBCfg[i].Snapshot != "" {
-			c.SourceDBCfg[i].Snapshot = strconv.Quote(c.SourceDBCfg[i].Snapshot)
+
+		if len(c.DMTask) == 0 {
+			log.Error("must set the `dm-task` if set `dm-addr`")
+			return false
 		}
 	}
 
-	if c.TargetDBCfg.InstanceID == "" {
-		c.TargetDBCfg.InstanceID = "target"
-	}
-	if c.TargetDBCfg.Snapshot != "" {
-		c.TargetDBCfg.Snapshot = strconv.Quote(c.TargetDBCfg.Snapshot)
-	}
-	if _, ok := sourceInstanceMap[c.TargetDBCfg.InstanceID]; ok {
-		log.Error("target has same instance id in source", zap.String("instance id", c.TargetDBCfg.InstanceID))
-		return false
-	}
+	// source DB, target DB and check table's information will get from DM, don't need to check them
+	if len(c.DMAddr) == 0 {
+		if len(c.SourceDBCfg) == 0 {
```

Comment:
Should we need to log something if `len(c.DMAddr) != 0` and `len(c.SourceDBCfg) != 0` (or some check for `c.TargetDBCfg`)?

Reply:
add in https://github.com/pingcap/tidb-tools/pull/377/commits/e3bb4bf891d6ddb50b9f53e0b5e0f5b7184cc7f0